This crate fails to build if the serde derive feature is enabled.

### I tried this:

I was trying to use this crate while also importing serde with the derive feature enabled.

### This happened:

The simplest reproduction is running `cargo build --workspace --all-targets --features "serde/derive"` which returns these errors

```
error[E0252]: the name `Deserialize` is defined multiple times
 --> core/src/models/map.rs:2:20
  |
1 | use serde::{Serialize, Deserialize};
  |                        ----------- previous import of the macro `Deserialize` here
2 | use serde_derive::{Deserialize, Serialize};
  |                    ^^^^^^^^^^^--
  |                    |
  |                    `Deserialize` reimported here
  |                    help: remove unnecessary import
  |
  = note: `Deserialize` must be defined only once in the macro namespace of this module

error[E0252]: the name `Serialize` is defined multiple times
 --> core/src/models/map.rs:2:33
  |
1 | use serde::{Serialize, Deserialize};
  |             --------- previous import of the macro `Serialize` here
2 | use serde_derive::{Deserialize, Serialize};
  |                                 ^^^^^^^^^ `Serialize` reimported here
  |
  = note: `Serialize` must be defined only once in the macro namespace of this module
```

### I expected this:

_No response_

### Is there a workaround?

One way to fix this is to fully qualify all the instances of Serialize and Deserialize in core/src/models/map.rs as serde::Serialize or serde_derive::Serialize.

### Anything else?

`cargo build --workspace --all-targets --features "serde/derive"` could be added to the github actions to check for this bug.

### Platform(s)

_No response_

### Community Notes

<!-- Please keep this note for the community -->
* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.name: Bug Report 🐞

--- a/.github/workflows/pipeline.yml
+++ b/.github/workflows/pipeline.yml
@@ -45,7 +45,7 @@ jobs:
           ${{ runner.os }}-cargo-build-
 
     - name: Build the project
-      run: cargo build --workspace --all-targets
+      run: cargo build --workspace --all-targets --features "serde/derive"
 
     - name: Run tests
       run: cargo test --workspace --all-targets

--- a/builders/Cargo.toml
+++ b/builders/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "serverless_workflow_builders"
-version = "1.0.0-alpha6.1"
+version = "1.0.0-alpha6.2"
 edition = "2021"
 authors = ["The Serverless Workflow Authors <cncf.serverless.workflow@gmail.com>"]
 description = "Contains services used to build ServerlessWorkflow workflow definitions programatically"
@@ -12,6 +12,6 @@ keywords = ["serverless-workflow", "sdk", "builders"]
 categories = ["config", "parsing", "data-structures", "api-bindings"]
 
 [dependencies]
-serverless_workflow_core = { path = "../core", version = "1.0.0-alpha6.1" }
+serverless_workflow_core = { path = "../core", version = "1.0.0-alpha6.2" }
 serde_json = "1.0" 
 serde_yaml = "0.9"

--- a/core/Cargo.toml
+++ b/core/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "serverless_workflow_core"
-version = "1.0.0-alpha6.1"
+version = "1.0.0-alpha6.2"
 edition = "2021"
 authors = ["The Serverless Workflow Authors <cncf.serverless.workflow@gmail.com>"]
 description = "Contains Serverless Workflow DSL models"

--- a/core/src/models/map.rs
+++ b/core/src/models/map.rs
@@ -1,10 +1,9 @@
 use serde::{Serialize, Deserialize};
-use serde_derive::{Deserialize, Serialize};
 use std::collections::HashMap;
 use std::hash::Hash;
 
 /// Represents an ordered key/value map array
-#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[derive(Debug, Clone, PartialEq, serde_derive::Serialize, serde_derive::Deserialize)]
 #[serde(bound(
     serialize = "TKey: Serialize, TValue: Serialize",
     deserialize = "TKey: for<'d> Deserialize<'d> + Eq + Hash + Clone + PartialEq, TValue: for<'d> Deserialize<'d> + Clone + PartialEq"

